File: src/dataset/python/difference-of-squares_exercism-ThomasZumsteg/gemini/GeminiFlash_difference_of_squares_v1.py
"""Function for calculating squares of sums or sums
of squares and differences between the two
"""
def square_of_sum(num):
    """Square of sum of numbers less then or equal to [num]"""
    sum_val = num * (num + 1) // 2
    return sum_val * sum_val

def sum_of_squares(num):
    """Sum of square of numbers less then or equal [num]"""
    return num * (num + 1) * (2 * num + 1) // 6

def difference(num):
    """Difference between the square_of_sums and sum_of_square"""
    return (num * (num + 1) // 2)**2 - num * (num + 1) * (2 * num + 1) // 6

File: src/dataset/javascript/difference-of-squares_exercism-ThomasZumsteg/gemini/GeminiFlash_difference-of-squares_v1.js
var Squares = function(num) {
  this.squareOfSums = (() => {
    const sum = (num * (num + 1)) / 2;
    return sum * sum;
  })();

  this.sumOfSquares = (num * (num + 1) * (2 * num + 1)) / 6;
  this.difference = this.squareOfSums - this.sumOfSquares;
};

module.exports = Squares;